### ISSUE TITLE
Subprocess crashes not reported within blocks

### ISSUE THREAD

**Issue by mqudsi** [fish-shell/fish-shell#4962]
```fish
~> foo
fish: “foo…” terminated by signal SIGABRT (Abort)
~> begin; foo; end;
~>
```

Gave me a hell of a time trying to understand why my script was failing without any visible errors.

**Comment by faho** [comment]
This seems to have been originally added on purpose. See parse_execution.cpp:run_1_job():

```c++
    job->set_flag(JOB_SKIP_NOTIFICATION,
                  is_subshell || is_block || is_event || !shell_is_interactive());
```

It seems safe to remove blocks from this, though I'd probably even report _crashes_ (SIGABRT) for events and "subshell"s (this is talking about command substitutions).

**Comment by faho** [comment]
Yeah, this was added way back in 2005, in 838ba08eaa883026d6fb88cd65dde33b16a564fd.

With the full commit message of "Event handling and job reaping updates".

**Comment by faho** [comment]
Bash's policy here seems to be to notify about crashes except in command substitutions - even if the shell isn't interactive!

So that would be 

```c++
    job->set_flag(JOB_SKIP_NOTIFICATION,
                  is_subshell);
```

Which I don't quite get.

Anyone got any ideas for a better policy than what we have now?

**Comment by faho** [comment]
So.... how about we just always report crashes?

I'm not entirely sure how to define "crash", though. My best attempt is "everything that was killed via one of the following signals:  SIGABRT, SIGBUS, SIGFPE, SIGILL, SIGSEGV, SIGSYS".

**Comment by mqudsi** [comment]
I would be on board with that.
